src/engine/enginebuffer.cpp

```diff
@@ -1136,7 +1151,7 @@ void EngineBuffer::processSlip(int iBufferSize) {
     bool enabled = static_cast<bool>(load_atomic(m_slipEnabled));
     if (enabled != m_bSlipEnabledProcessing) {
         m_bSlipEnabledProcessing = enabled;
-        if (enabled) {
+        if (enabled || m_slipCancelled.fetchAndStoreAcquire(0)) {
```

Comment:
I think we don't need a variable at all.  Just check the state of the slipcancel control object.  if true, cancel.  Then you can delete the handler and the connection too.